### PR TITLE
gnrc_icmpv6_error: consider originator's destination address correctly

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
While 485dbd1fdac9c965c1f734b1dda711b35bb4ac1a (from #12175) was right in assuming that the for most ICMPv6 error messages the originating packet's destination address must not be a multicast, this is not the case for _all_ ICMPv6 error messages (see [RFC 4443], section 2.4(e.3)). Additionally, 485dbd1fdac9c965c1f734b1dda711b35bb4ac1a removed the check for the source address ([RFC 4443], section 2.4(e.6)), which this PR re-adds.

[RFC 4443]: https://tools.ietf.org/html/rfc4443#section-2.4

### Testing procedure
- Read the RFC and confirm, that the rules are applied correctly now.
- [Release specifications for ICMPv6 errors should still pass](https://github.com/RIOT-OS/Release-Specs/blob/master/10-icmpv6-error/10-icmpv6-error.md).

### Issues/PRs references
Accompanying #12440, but not necessary to fix the problem there

### CODE REVIEW

**Comment by miri64** [comment]
While it would definitely be good to have this in the release, it might not be critical enough for some to justify a backport. This is why I marked the impact as minor (also because it only just touches this one module).

**Comment by miri64** [comment]
Fixed error in both PR title and commit summary.

File: sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c [hunk 1]
```diff
@@ -234,24 +234,36 @@ static void _send(gnrc_pktsnip_t *pkt, const gnrc_pktsnip_t *orig_pkt,
...
     }
+    /* ipv6_hdr->dst may only be multicast for Packet Too Big Messages
+     * and Parameter Problem Messages with code 2, see
+     * https://tools.ietf.org/html/rfc4443#section-2.4 (e.3) */
+    else if (ipv6_addr_is_multicast(&ipv6_hdr->dst)) {
+        if ((type != ICMPV6_PKT_TOO_BIG) ||
+            ((type == ICMPV6_PARAM_PROB) &&
+             (code != ICMPV6_ERROR_PARAM_PROB_OPT))) {
+            ipv6 = NULL;
+        }
```
Comment (benpicco) [inline sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: This condition is always true if `type != ICMPV6_PKT_TOO_BIG`.

Comment (miri64) [reply sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: Yepp, that's the point. From [RFC 4443, section 2.4](https://tools.ietf.org/html/rfc4443#section-2.4) (`ipv6 == NULL` will result in the sending functions being aborted, see code below):

>     (e) An ICMPv6 error message MUST NOT be originated as a result of
>         receiving the following:
>          […]
>          (e.3) A packet destined to an IPv6 multicast address.  (There are
>                two exceptions to this rule: (1) the Packet Too Big Message
>                (Section 3.2) to allow Path MTU discovery to work for IPv6
>                multicast, and (2) the Parameter Problem Message, Code 2
>                (Section 3.4) reporting an unrecognized IPv6 option (see
>                Section 4.2 of [IPv6]) that has the Option Type highest-
>                order two bits set to 10).

Comment (benpicco) [reply sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: But it's also true if `type == ICMPV6_PARAM_PROB && code == ICMPV6_ERROR_PARAM_PROB_OPT`

Comment (benpicco) [reply sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: `type != ICMPV6_PKT_TOO_BIG` already evaluates to `true`, so it's irrelevant what's in the second `or` case.

Comment (miri64) [reply sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: You are right. I did not find time to address this yet though. Will fix ASAP.

Comment (miri64) [reply sys/net/gnrc/network_layer/icmpv6/error/gnrc_icmpv6_error.c]: Should be fixed now, but I did not retest yet.

**Comment by miri64** [comment]
Ok, retested with the release specs but also tried to trigger a Packet Too Big message for packet addressed to multicast (works). We don't send currently Parameter Problem Message, Code 2, so I was unable to test.

**Comment by miri64** [comment]
(for more information on the latter see https://tools.ietf.org/html/rfc8200#section-4.2. We don't support neither Hop-By-Hop-options nor Destination options at the moment)

**Comment by benpicco** [comment]
Looks good now, please squash!

**Comment by miri64** [comment]
Squashed

**Comment by miri64** [comment]
Backport provided in #12506

**Pull request merged**